buildscripts/builddrivers.py

```diff
@@ -0,0 +1,291 @@
+        root_dir = 'C:' + os.sep
+        
+        quit = False
+        while not quit:
+            if not self.rebuild and self.local:
+                self.clean_or_remove(root_dir, work_dir)
+                
+            logfile = self.util.get_logfile_name()
+
+            try:
+                dest = None
+                if self.local:
+                    dest = root_dir
+                    
+                self.build_extensions(dest, logfile)
+                print('Build Completed')
+            except:
+                print('Something went wrong. Build incomplete.')
+                if self.local:          # display log file only when building locally
+                    os.startfile(os.path.join(root_dir, 'php-sdk', logfile))
+                os.chdir(work_dir)    
+                break
+
+            # Only ask when building locally
+            if self.local:               
+                choice = input("Rebuild the same configuration(yes) or quit (no) [yes/no]: ")
+
+                if choice.lower() == 'yes' or choice.lower() == 'y' or choice.lower() == '':
+                    print('Rebuilding drivers...')
+                    self.make_clean = False
+                    self.rebuild = True
+                    self.util.remove_prev_build(root_dir)
+                else:
+                    quit = True
+            else:
+                quit = True
+            
+            os.chdir(work_dir)    
+
+def validate_input(question, values):
+    """Return the user selected value, and it must be valid based on *values*."""
+    while True:
+        options = values.split('/') 
+        prompt = '[' + values + ']'
+        value = input(question + prompt + ': ')
+        value = value.lower()
+        if not value in options:
+            print("An invalid choice is entered. Choose from", prompt)
+        else:
+            break
+    return value
+
+################################### Main Function ###################################
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--PHPVER', help="PHP version, e.g. 7.1.*, 7.2.* etc.")
+    parser.add_argument('-a', '--ARCH', choices=['x64', 'x86'])
+    parser.add_argument('-t', '--THREAD', choices=['nts', 'ts'])
+    parser.add_argument('-d', '--DRIVER', choices=['all', 'sqlsrv', 'pdo_sqlsrv'])
+    parser.add_argument('-m', '--DEBUG', default='no', choices=['yes', 'no'], help="enable debug mode")
+    parser.add_argument('-r', '--REPO', default='Microsoft', help="GitHub repository")
+    parser.add_argument('-b', '--BRANCH', default='dev', help="GitHub repository branch")
+    parser.add_argument('-g', '--GITHUB', default='yes', help="get source from GitHub or not")
+    parser.add_argument('-k', '--PACKAGE', default='Latest', help="the package name for the drivers")
+    parser.add_argument('-p', '--PATH', default=None, help="the remote destination for the drivers (do not use this when building locally)")
```

Comment:
DESTINATION_PATH (or DESTPATH) would be more descriptive as path could be any old path. It's also not necessarily "remote". And I MAY choose to do this when "building locally" that's up to me to decide how to use this script